addons/point_of_sale/models/pos_session.py

```diff
@@ -254,10 +254,12 @@ def unlink(self):
 
     def login(self):
         self.ensure_one()
-        login_number = self.login_number + 1
-        self.write({
-            'login_number': login_number,
-        })
+        db_name = self._cr.dbname
+        db_registry = registry(db_name)
+        with db_registry.cursor() as cr:
+            cr.execute('SELECT login_number FROM pos_session WHERE id = %s FOR UPDATE NOWAIT', (self.id,))
```

Comment:
instead of hardcoding pos_session in the query, use self._table (or name) and pass as query param.

Also, leave a quick explanation why the need for new cursor, and not the existing one.